File: lib/core/constants/calculation_contract.dart
// THIS FILE IS THE SINGLE SOURCE OF TRUTH - mirror of CalculationContract.kt
// Do not change values without updating the mirror file in android/app/src/main/kotlin/com/qada/fard/prayer/CalculationContract.kt
abstract class CalculationContract {
  // Calculation Methods
  static const int methodMuslimWorldLeague = 0;
  static const int methodEgyptian = 1;
  static const int methodKarachi = 2;
  static const int methodUmmAlQura = 3;
  static const int methodDubai = 4;
  static const int methodMoonSightingCommittee = 5;
  static const int methodNorthAmerica = 6;
  static const int methodKuwait = 7;
  static const int methodQatar = 8;
  static const int methodSingapore = 9;
  static const int methodTehran = 10;
  static const int methodTurkey = 11;

  // Madhabs
  static const int madhabShafi = 0;
  static const int madhabHanafi = 1;

  // High Latitude Rules
  static const int highLatMiddleOfTheNight = 0;
  static const int highLatSeventhOfTheNight = 1;
  static const int highLatTwilightAngle = 2;

  // Channel & Pref Keys
  static const String channelName = 'com.qada.fard/instant_updates';
  static const String prefPrefix = 'flutter.';
}

File: android/app/src/main/kotlin/com/qada/fard/prayer/CalculationContract.kt
package com.qada.fard.prayer

/**
 * THIS FILE IS THE SINGLE SOURCE OF TRUTH - mirror of calculation_contract.dart
 * Do not change values without updating the mirror file in lib/core/constants/calculation_contract.dart
 */
object CalculationContract {
    // Calculation Methods
    const val METHOD_MUSLIM_WORLD_LEAGUE = 0
    const val METHOD_EGYPTIAN = 1
    const val METHOD_KARACHI = 2
    const val METHOD_UMM_AL_QURA = 3
    const val METHOD_DUBAI = 4
    const val METHOD_MOON_SIGHTING_COMMITTEE = 5
    const val METHOD_NORTH_AMERICA = 6
    const val METHOD_KUWAIT = 7
    const val METHOD_QATAR = 8
    const val METHOD_SINGAPORE = 9
    const val METHOD_TEHRAN = 10
    const val METHOD_TURKEY = 11
    const val METHOD_USER_CUSTOM = 12

    // Madhabs
    const val MADHAB_SHAFI = 0
    const val MADHAB_HANAFI = 1

    // High Latitude Rules
    const val HIGH_LAT_MIDDLE_OF_THE_NIGHT = 0
    const val HIGH_LAT_SEVENTH_OF_THE_NIGHT = 1
    const val HIGH_LAT_TWILIGHT_ANGLE = 2

    // Channel & Pref Keys
    const val CHANNEL_NAME = "com.qada.fard/instant_updates"
    const val PREF_PREFIX = "flutter."
}
